### DISCUSSION TITLE
On start mc-image-helper fails with timeout

### DISCUSSION THREAD

**Discussion by johannes-wolf** [itzg/docker-minecraft-server#2466] [Q&A]
### Describe the problem

When starting the container, the `mc-image-helper` fails with the following message:
```
[mc-image-helper] 15:22:09.641 ERROR : 'resolve-minecraft-version' command failed. Version is 1.36.5
mc_1  | reactor.core.Exceptions$ReactiveException: io.netty.channel.ConnectTimeoutException: connection timed out: launchermeta.mojang.com/13.107.246.45:443
```

### Container definition

```
version: "3.8"

services:
  mc:
    image: itzg/minecraft-server
    tty: true
    stdin_open: true
    ports:
      - "25565:25565"
    environment:
      EULA: "TRUE"
      MEMORY: "2G"
      VERSION: "1.20.2"
      WHITELIST_FILE: "/data/whitelist"
      OPS_FILE: "/data/admins"
    volumes:
      - ./data:/data
```

### Container logs

```
mc_1  | [init] Running as uid=1000 gid=1000 with /data as 'drwxr-xr-x 3 1000 1000 4096 Nov  3 15:07 /data'
mc_1  | [init] Resolving type given VANILLA
mc_1  | [mc-image-helper] 15:22:09.641 ERROR : 'resolve-minecraft-version' command failed. Version is 1.36.5
mc_1  | reactor.core.Exceptions$ReactiveException: io.netty.channel.ConnectTimeoutException: connection timed out: launchermeta.mojang.com/13.107.246.45:443
mc_1  |         at reactor.core.Exceptions.propagate(Exceptions.java:408)
mc_1  |         at reactor.core.publisher.BlockingSingleSubscriber.blockingGet(BlockingSingleSubscriber.java:101)
mc_1  |         at reactor.core.publisher.Mono.block(Mono.java:1712)
mc_1  |         at me.itzg.helpers.versions.ResolveMinecraftVersionCommand.call(ResolveMinecraftVersionCommand.java:26)
mc_1  |         at me.itzg.helpers.versions.ResolveMinecraftVersionCommand.call(ResolveMinecraftVersionCommand.java:12)
mc_1  |         at picocli.CommandLine.executeUserObject(CommandLine.java:2041)
mc_1  |         at picocli.CommandLine.access$1500(CommandLine.java:148)
mc_1  |         at picocli.CommandLine$RunLast.executeUserObjectOfLastSubcommandWithSameParent(CommandLine.java:2461)
mc_1  |         at picocli.CommandLine$RunLast.handle(CommandLine.java:2453)
mc_1  |         at picocli.CommandLine$RunLast.handle(CommandLine.java:2415)
mc_1  |         at picocli.CommandLine$AbstractParseResultHandler.execute(CommandLine.java:2273)
mc_1  |         at picocli.CommandLine$RunLast.execute(CommandLine.java:2417)
mc_1  |         at picocli.CommandLine.execute(CommandLine.java:2170)
mc_1  |         at me.itzg.helpers.McImageHelper.main(McImageHelper.java:157)
mc_1  |         Suppressed: java.lang.Exception: #block terminated with an error
mc_1  |                 at reactor.core.publisher.BlockingSingleSubscriber.blockingGet(BlockingSingleSubscriber.java:103)
mc_1  |                 ... 12 common frames omitted
mc_1  | Caused by: io.netty.channel.ConnectTimeoutException: connection timed out: launchermeta.mojang.com/13.107.246.45:443
mc_1  |         at io.netty.channel.nio.AbstractNioChannel$AbstractNioUnsafe$1.run(AbstractNioChannel.java:261)
mc_1  |         at io.netty.util.concurrent.PromiseTask.runTask(PromiseTask.java:98)
mc_1  |         at io.netty.util.concurrent.ScheduledFutureTask.run(ScheduledFutureTask.java:153)
mc_1  |         at io.netty.util.concurrent.AbstractEventExecutor.runTask(AbstractEventExecutor.java:173)
mc_1  |         at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:166)
mc_1  |         at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:470)
mc_1  |         at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:569)
mc_1  |         at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997)
mc_1  |         at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
mc_1  |         at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
mc_1  |         at java.base/java.lang.Thread.run(Unknown Source)

```

**Comment by dlabreu** [comment]
+1 here same issue.

**Comment by johannes-wolf** [comment]
It is a routing issue on the host side, the container in unable to ping any host. I re-opened the issue because other people have the same problem.

**Comment by itzg** [comment]
Let's not reopen an issue I can't fix...but I will convert to a discussion so it's more visible.

**Comment by Rick-O-Shay** [comment]
Hi,

I am new to GitHub and also docker containers. I am trying to setup a minecraft server on my Synology NAS using a docker container and I found this thread after googling because I have the exact same issue. The OP was made just hours ago. Is this something wrong in our setup or is there another issue at hand here? I tried vanilla and paper, and both are just erroring out on me and will not run.

Any info would be appreciated for a noob like me.

Thanks,

**Reply by Rick-O-Shay** [reply]
Sorry, going nuts here. Does this post mean that we have config properly, but there is a problem outside of what we have done? Is every self-hosted minecraft server down??

**Reply by itzg** [reply]
Yes, the posted container configuration is fine. The container starts fine for me, apparently for most others since I see no mention of networking issues on the Discord server, and even the OP from @johannes-wolf since they reported it was a routing issue on their host.

Perhaps @johannes-wolf you can share more details about how you resolved your issue.

**Reply by Rick-O-Shay** [reply]
Interesting. I disabled my firewall in my router for a split second to test if that was the issue and it works fine, so apparently there is something other than the default port of 25565 that needs to be opened up, but now I have to troubleshoot that.... 
 Thanks

**Reply by johannes-wolf** [reply]
I could not get it to work, and my routing knowledge is very limited. My “solution” was to host the Minecraft server without using docker - without this project. This container creates a new network and `ip route` also looks fine to my understanding…

**Comment by guercheLE** [comment]
Getting same error. If take TYPE=PAPER, it goes fine and downloads VANILLA. How do I figure out which URL the script is trying to download and to fix it?

28/04/2024
13:36:21
[init] Running as uid=1000 gid=1000 with /data as 'drwxrwxr-x 12 1000 1000 4096 Nov 26 12:16 /data'
28/04/2024
13:36:21
[init] Resolving type given PAPER
28/04/2024
13:37:53
[mc-image-helper] 13:37:53.484 ERROR : 'install-paper' command failed. Version is 1.38.13
28/04/2024
13:37:53
reactor.core.Exceptions$ReactiveException: io.netty.channel.ConnectTimeoutException: connection timed out after 30000 ms: api.papermc.io/172.67.72.198:443
28/04/2024
13:37:53
	at reactor.core.Exceptions.propagate(Exceptions.java:408)
28/04/2024
13:37:53
	at reactor.core.publisher.BlockingSingleSubscriber.blockingGet(BlockingSingleSubscriber.java:101)
28/04/2024
13:37:53
	at reactor.core.publisher.Mono.block(Mono.java:1712)
28/04/2024
13:37:53
	at me.itzg.helpers.paper.InstallPaperCommand.useCoordinates(InstallPaperCommand.java:153)
28/04/2024
13:37:53
	at me.itzg.helpers.paper.InstallPaperCommand.call(InstallPaperCommand.java:110)
28/04/2024
13:37:53
	at me.itzg.helpers.paper.InstallPaperCommand.call(InstallPaperCommand.java:35)
28/04/2024
13:37:53
	at picocli.CommandLine.executeUserObject(CommandLine.java:2041)
28/04/2024
13:37:53
	at picocli.CommandLine.access$1500(CommandLine.java:148)
28/04/2024
13:37:53
	at picocli.CommandLine$RunLast.executeUserObjectOfLastSubcommandWithSameParent(CommandLine.java:2461)
28/04/2024
13:37:53
	at picocli.CommandLine$RunLast.handle(CommandLine.java:2453)
28/04/2024
13:37:53
	at picocli.CommandLine$RunLast.handle(CommandLine.java:2415)
28/04/2024
13:37:53
	at picocli.CommandLine$AbstractParseResultHandler.execute(CommandLine.java:2273)
28/04/2024
13:37:53
	at picocli.CommandLine$RunLast.execute(CommandLine.java:2417)
28/04/2024
13:37:53
	at picocli.CommandLine.execute(CommandLine.java:2170)
28/04/2024
13:37:53
	at me.itzg.helpers.McImageHelper.main(McImageHelper.java:159)
28/04/2024
13:37:53
	Suppressed: java.lang.Exception: #block terminated with an error
28/04/2024
13:37:53
		at reactor.core.publisher.BlockingSingleSubscriber.blockingGet(BlockingSingleSubscriber.java:103)
28/04/2024
13:37:53
		... 13 common frames omitted
28/04/2024
13:37:53
Caused by: io.netty.channel.ConnectTimeoutException: connection timed out after 30000 ms: api.papermc.io/172.67.72.198:443
28/04/2024
13:37:53
	at io.netty.channel.epoll.AbstractEpollChannel$AbstractEpollUnsafe$2.run(AbstractEpollChannel.java:615)
28/04/2024
13:37:53
	at io.netty.util.concurrent.PromiseTask.runTask(PromiseTask.java:98)
28/04/2024
13:37:53
	at io.netty.util.concurrent.ScheduledFutureTask.run(ScheduledFutureTask.java:153)
28/04/2024
13:37:53
	at io.netty.util.concurrent.AbstractEventExecutor.runTask(AbstractEventExecutor.java:173)
28/04/2024
13:37:53
	at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:166)
28/04/2024
13:37:53
	at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:470)
28/04/2024
13:37:53
	at io.netty.channel.epoll.EpollEventLoop.run(EpollEventLoop.java:416)
28/04/2024
13:37:53
	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997)
28/04/2024
13:37:53
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
28/04/2024
13:37:53
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
28/04/2024
13:37:53
	at java.base/java.lang.Thread.run(Unknown Source)
28/04/2024
13:37:53
[init] ERROR: failed to download paper

**Reply by guercheLE** [reply]
The following command:

curl -X 'GET' \
  'https://api.papermc.io/v2/projects/paper/versions/1.20.1/builds/196/downloads/234a9b32098100c6fc116664d64e36ccdb58b5b649af0f80bcccb08b0255eaea' \
  -H 'accept: application/json'
  
  Is returning:
  
  {
  "error": "Download not found."
}

**Reply by guercheLE** [reply]
Same for the following command:

curl -X 'GET' \
  'https://api.papermc.io/v2/projects/paper/versions/1.20.5/builds/15/downloads/f7e96c63895aae25188f907844ad151c4ad2c620e002f16c446a58b4c32e9161' \
  -H 'accept: application/json'

**Comment by mitchjb2008** [comment]
I fixed this issue on my server by adding `network_mode: "host"` to the docker compose file:

```
services:
  minecraft:
    image: itzg/minecraft-server
    network_mode: "host"
```

**Reply by SanjayVikash** [reply]
Yep this works

**Reply by templeofshadow** [reply]
FYI, for anyone who uses the itzg/mc-backup sidecar: in my admittedly limited testing, using this fix (network_mode: host) will break the backup functionality. I've yet to find a fix, and will probably switch to using other backup methods. 

For itzg (or any other maintainers), this is my log for the mc-backup container:
```
time="2025-04-27T15:57:27Z" level=info msg="Skipping uid/gid change since current user is not root"

2025-04-27T15:57:27+0000 INFO waiting initial delay of 0...

2025-04-27T15:57:27+0000 INFO waiting for rcon readiness...

2025-04-27T15:57:27+0000 ERROR Unable to execute rcon-cli save-on - try 0/5. Retrying in 10s

2025-04-27T15:57:27+0000 ERROR Failure reason: 2025/04/27 15:57:27 Failed to connect to RCON serverdial tcp: lookup mc on 192.168.2.1:53: no such host	

2025-04-27T15:57:37+0000 ERROR Unable to execute rcon-cli save-on - try 1/5. Retrying in 10s

2025-04-27T15:57:37+0000 ERROR Failure reason: 2025/04/27 15:57:37 Failed to connect to RCON serverdial tcp: lookup mc on 192.168.2.1:53: no such host	

2025-04-27T15:57:47+0000 ERROR Unable to execute rcon-cli save-on - try 2/5. Retrying in 10s

2025-04-27T15:57:47+0000 ERROR Failure reason: 2025/04/27 15:57:47 Failed to connect to RCON serverdial tcp: lookup mc on 192.168.2.1:53: no such host	

2025-04-27T15:57:57+0000 ERROR Unable to execute rcon-cli save-on - try 3/5. Retrying in 10s

2025-04-27T15:57:57+0000 ERROR Failure reason: 2025/04/27 15:57:57 Failed to connect to RCON serverdial tcp: lookup mc on 192.168.2.1:53: no such host	

2025-04-27T15:58:07+0000 ERROR Unable to execute rcon-cli save-on - try 4/5. Retrying in 10s

2025-04-27T15:58:07+0000 ERROR Failure reason: 2025/04/27 15:58:07 Failed to connect to RCON serverdial tcp: lookup mc on 192.168.2.1:53: no such host	

2025-04-27T15:58:17+0000 ERROR Unable to execute rcon-cli save-on - try 5/5. Retrying in 10s

2025-04-27T15:58:17+0000 ERROR Failure reason: 2025/04/27 15:58:17 Failed to connect to RCON serverdial tcp: lookup mc on 192.168.2.1:53: no such host	
```

My setup is pretty atypical, so just in case that's a factor, I'll list it below.
Main server rig is a computer with a i7-8700k and 32GB ram, running proxmox. On proxmox, there is a opnsense router VM acting as a main gateway and firewall to all other VMs. Then, attached to the private network behind the gateway is a docker host VM which is running my docker containers (including my mc servers). For container management I'm using portainer.

EDIT: Found a fix. In the environment section of the compose file for itzg/minecraft-server container:
1. For a multi-server setup, make sure to set a RCON_PORT in the different from the default and from the other servers
2. Also, set a different port to default for SERVER_PORT

In the section of the compose file for itzg/mc-backup container:
1. Make sure to set RCON_HOST to localhost
2. Set RCON_PORT to the same port as the above.
3. Set SERVER_PORT to the same as above

Generally: SET network_mode: host FOR ALL CONTAINERS IN THE COMPOSE FILE.

**Comment by Sk8Gnarley** [comment]
I fixed this by deleting server.properties and eula.txt so it would recreate them. I also made sure the permission on the files I dropped in _data/ were matching. Hope this helps anyone!
I have no idea why this worked, but it did.

**Comment by The1FarmerJohn** [comment]
Not sure if anyone may find this useful, but I had these same exact io.netty errors out of nowhere. Turns out my Tailscale key had expired (I have Tailscale installed on host system itself.) I have now corrected the key issue, restarted docker and all containers - everything works perfect now and containers deploy/start without errors

**Reply by The1FarmerJohn** [reply]
(I realize that my setup isn't optimal and could have configured tailscale via different methods, but it's what worked for me at the time with my limited knowledge and I'm too lazy to change anything now)

**Comment by seewood** [comment]
hey to the minecraft server guys and thanks to @itzg for the work you put in.

Im getting an issue starting my minecraft server on my synology container manager (docker). Its crashing right after launch. and im getting a very long log but the start of the log is shown here:
<img width="864" height="150" alt="image" src="https://github.com/user-attachments/assets/5fbc0953-dd01-462c-bcec-d745b731b7c7" />

someone know what im missing or doing wrong? Ive set the container up with 3 similar YouTube tutorials, but end up always with a crashing setup.

**Reply by itzg** [reply]
Yours is a DNS issue. A DNS server such as 1.1.1.1, can be specified per container but I'll have to look on my Synology to see where.

**Reply by itzg** [reply]
Looks like it needs to be done via the projects / docker compose section

<img width="1014" height="405" alt="image" src="https://github.com/user-attachments/assets/6246b00e-834e-4e77-9243-13fcbb2e9728" />

With that can set DNS servers in the service section

https://docs.docker.com/reference/compose-file/services/#dns

**Reply by seewood** [reply]
oh that was fast, so I simply add this to my compose file?

**Reply by seewood** [reply]
ok this worked for me. Can you go in detail why DNS is needed for me? and how did you see thats a DNS issue btw ? :)

**Reply by itzg** [reply]
@seewood the first line of your log says "failed to resolve". As for why, I'm not sure. DNS is supposed to be handled by the docker setup

https://docs.docker.com/engine/network/#dns-services

so something is wrong with your docker installation. When people encounter this it's easier to set the container dns rather than diagnose the root cause.

**Reply by seewood** [reply]
oh, i understand that :) ok... im having issues to get the port forwarding to work so i checked out a few sources on the internet and some say you need to set network_mode : host - this ends up again in the same issue as before. its crashing and the same msg appears. 

ive removed the dns settings, obviously when im host.

This is my setup:

<img width="315" height="421" alt="Bildschirmfoto 2026-02-11 um 15 35 02" src="https://github.com/user-attachments/assets/df2dda82-9d53-47a1-ab88-7f5f199c945d" />

**Reply by itzg** [reply]
I'm afraid using host networking mode is a last resort when also the docker installation is broken and not properly configuring the iptables firewall. 

I would suggest reinstalling docker since something is broken with its networking.

**Reply by seewood** [reply]
hmm... ive just installed it once and never used it since now. I will check that.

**Reply by seewood** [reply]
any chance to get in touch with you on discord?

**Reply by itzg** [reply]
Community support available at

https://discord.gg/DXfKpjB
